### PR TITLE
refactor: simplified example tests

### LINKED ISSUE #864: Simplify example tests
#### Reason/Context
The examples always run the [generate function](https://github.com/asyncapi/modelina/blob/527cc6a84f6ae5229d9b352fcbfc4b0ad8249036/examples/adapting-input-and-output/index.ts#L54), It is, however, a problem from a testing point of view the [models are generated twice](https://github.com/asyncapi/modelina/blob/527cc6a84f6ae5229d9b352fcbfc4b0ad8249036/examples/adapting-input-and-output/index.spec.ts#L11).

Instead, it should only be generated once.

To fix this we need to add the following wrapper around each generate call in the example index file:
```ts
if (require.main === module) {
  generate();
}
```
This means that it only generates the models if run from the CLI (i.e. `npm run start`).

We can then adapt the tests to only be called 1x instead of 2x. This means [changing this](https://github.com/asyncapi/modelina/blob/527cc6a84f6ae5229d9b352fcbfc4b0ad8249036/examples/adapting-input-and-output/index.spec.ts#L9) to:
```ts
    await generate();
    expect(spy.mock.calls.length).toEqual(1);
    expect(spy.mock.calls[0]).toMatchSnapshot();
```

### PR DESCRIPTION
**Pull request by nitintejuja**
**Description**

- Simplified adapting-input-and-output index.ts file as `generate` function was getting called 2 times and updated spec files accordingly.

**Related issue(s)**
Fixes #864

### CODE REVIEW

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Hey @nitintejuja, the issue that is referred is actually to change this for all the examples 🙂 Might not have been specified that well enough 🤔 Do you wish to apply the change to the rest of the examples, or just this one?

Also, please make sure you target the `next` branch and create your branch from it, otherwise, you will get a bunch of conflicts to solve 🙂 

Let me know if you have any problems.

**Comment by nitintejuja** [comment]
@jonaslagoni i'm facing issue in 2 tests . I have made required changes but its not working .
below are testfiles
`examples/typescript-generate-comments/index.spec.ts`
`examples/typescript-enum-type/index.spec.ts`

can you help me ?

**Review by jonaslagoni — COMMENTED** [review]
Finally back from holiday, let me check @nitintejuja!

File: examples/typescript-enum-type/index.spec.ts [hunk 1]
```diff
@@ -9,15 +9,15 @@ describe('Should be able to render correct enums based on options', () => {
   });
   test('and should log expected output to console', async function testIt(): Promise<void> {
     // Wait for first call to the `generate` function is over, and we can reliably know the output
-    if (spy.mock.calls.length !== 6) {
+    if (spy.mock.calls.length !== 3) {
       await timeout(50);
       return testIt();
     }
```
Comment (jonaslagoni) [inline examples/typescript-enum-type/index.spec.ts:15]:
```suggestion
```

You can safely remove this 🙂

File: examples/generate-go-enums/index.spec.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ describe('Should be able to render Go Enums', () => {
   test('and should log expected output to console', async () => {
     await generate();
     //Generate is called 2x, so even though we expect 3 model (root and three different enums type), we double it
```
Comment (jonaslagoni) [inline examples/generate-go-enums/index.spec.ts:10]:
```suggestion
```

File: examples/java-from-typescript-type/index.spec.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ describe('Should be able to generate Java model from a TypeScript type file', ()
   test('and should log expected output to console', async () => {
     await generate();
     //Generate is called 4x, so even though we expect 1 model, we double it
```
Comment (jonaslagoni) [inline examples/java-from-typescript-type/index.spec.ts:10]:
```suggestion
```

File: examples/javascript-use-cjs/index.spec.ts [hunk 4]
```diff
@@ -8,8 +8,7 @@ describe('Should be able to render models to ESM module system', () => {
   test('and should log expected output to console', async () => {
     await generate();
     //Generate is called 2x, so even though we expect 2 models, we double it
```
Comment (jonaslagoni) [inline examples/javascript-use-cjs/index.spec.ts:10]:
```suggestion
```

File: examples/javascript-use-esm/index.spec.ts [hunk 5]
```diff
@@ -8,8 +8,7 @@ describe('Should be able to render models to ESM module system', () => {
   test('and should log expected output to console', async () => {
     await generate();
     //Generate is called 2x, so even though we expect 2 models, we double it
```
Comment (jonaslagoni) [inline examples/javascript-use-esm/index.spec.ts:10]:
```suggestion
```

File: examples/openapi-from-object/index.spec.ts [hunk 6]
```diff
@@ -7,8 +7,8 @@ describe('Should be able to process a pure OpenAPI object', () => {
   test('and should log expected output to console', async () => {
     await generate();
     //Generate is called 2x, so even though we expect 2 models, we double it
```
Comment (jonaslagoni) [inline examples/openapi-from-object/index.spec.ts:9]:
```suggestion
```

File: examples/typescript-use-cjs/index.spec.ts [hunk 7]
```diff
@@ -8,8 +8,7 @@ describe('Should be able to render models to ESM module system', () => {
   test('and should log expected output to console', async () => {
     await generate();
     //Generate is called 2x, so even though we expect 2 models, we double it
```
Comment (jonaslagoni) [inline examples/typescript-use-cjs/index.spec.ts:10]:
```suggestion
```

File: examples/typescript-use-esm/index.spec.ts [hunk 8]
```diff
@@ -8,8 +8,7 @@ describe('Should be able to render models to ESM module system', () => {
   test('and should log expected output to console', async () => {
     await generate();
     //Generate is called 2x, so even though we expect 2 models, we double it
```
Comment (jonaslagoni) [inline examples/typescript-use-esm/index.spec.ts:10]:
```suggestion
```

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
I think you just have to update the snapshots 🙂 You can do this by running `npm run test:examples -- -u`

File: examples/typescript-enum-type/index.spec.ts (hunk 1 above)
Comment (nitintejuja) [reply examples/typescript-enum-type/index.spec.ts]: i have removed this piece of code but still i'm getting failed test for this spec file. can you help me ?
Thanks

Comment (jonaslagoni) [reply examples/typescript-enum-type/index.spec.ts]: Have you updated the snapshots? `npm run test -- -u`

Comment (nitintejuja) [reply examples/typescript-enum-type/index.spec.ts]: yes but on snapshot creation it gives error.

Comment (jonaslagoni) [reply examples/typescript-enum-type/index.spec.ts]: Which error are you getting?

Comment (nitintejuja) [reply examples/typescript-enum-type/index.spec.ts]: im getting below error 
`FAIL examples/typescript-enum-type/index.spec.ts (18.144 s)
  ● Should be able to render correct enums based on options › and should log expected output to console

    expect(received).toMatchSnapshot()

    Snapshot name: `Should be able to render correct enums based on options and should log expected output to console 1`

    - Snapshot  - 14
    + Received  +  0

    @@ -1,19 +1,5 @@
    - Generator output with Union:
    - class Root {
    -   private _eventType?: EventType;
    -
    -   constructor(input: {
    -     eventType?: EventType,
    -   }) {
    -     this._eventType = input.eventType;
    -   }
    -
    -   get eventType(): EventType | undefined { return this._eventType; }
    -   set eventType(eventType: EventType | undefined) { this._eventType = eventType; }
    - }
    - type EventType = "ping" | "pong";
      Generator output with Enum:
      class Root {
        private _eventType?: EventType;

        constructor(input: {

      12 |     expect(spy.mock.calls.length).toEqual(6);
      13 |     const generatedContent = spy.mock.calls.slice(3).join('\n');
    > 14 |     expect(generatedContent).toMatchSnapshot();
         |                              ^
      15 |   });
      16 | });
      17 |

      at examples/typescript-enum-type/index.spec.ts:14:30
      at fulfilled (examples/typescript-enum-type/index.spec.ts:5:58)
`

File: examples/typescript-enum-type/index.spec.ts [hunk 9]
```diff
@@ -7,17 +7,10 @@ describe('Should be able to render correct enums based on options', () => {
...
-      return testIt();
-    }
-
+  test('and should log expected output to console', async (): Promise<void> => {
     await generate();
-    //Generate is called 2 times, so even though we expect 6 console logs we double it
-    expect(spy.mock.calls.length).toEqual(12);
-    const generatedContent = spy.mock.calls.slice(6).join('\n');
+    expect(spy.mock.calls.length).toEqual(6);
+    const generatedContent = spy.mock.calls.slice(3).join('\n');
```
Comment (jonaslagoni) [inline examples/typescript-enum-type/index.spec.ts:13]:
```suggestion
    const generatedContent = spy.mock.calls.join('\n');
```

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Ahh damn, found out the issue 🤦 I recently changed the test script which means that it is no longer possible to update the snapshots like this 🙃 

After applying the above comment you need to run the following command `./node_modules/.bin/jest  ./examples --testPathIgnorePatterns ./examples/integrate-with-react --testPathIgnorePatterns ./examples/TEMPLATE -u` to update all the snapshots...

So sorry for this!

File: examples/typescript-enum-type/index.spec.ts (hunk 1 above)
Comment (jonaslagoni) [reply examples/typescript-enum-type/index.spec.ts]: please see https://github.com/asyncapi/modelina/pull/868#pullrequestreview-1117781992

File: examples/typescript-enum-type/index.spec.ts (hunk 9 above)
Comment (nitintejuja) [reply examples/typescript-enum-type/index.spec.ts:13]: now its working fine and all testcases passed. Just one more doubt, should i commit latest snapshots ?

Comment (jonaslagoni) [reply examples/typescript-enum-type/index.spec.ts:13]: Yes please :)

Comment (nitintejuja) [reply examples/typescript-enum-type/index.spec.ts]: done :heavy_check_mark:

**Review by jonaslagoni — APPROVED** [review]

**Comment by jonaslagoni** [comment]
Thanks for the hard work @nitintejuja! This is such a great change!

@all-contributors please add @nitintejuja for test

**Pull request merged**